app/javascript/components/cloud-networks/create-nsxt-cloud-network-form.jsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import MiqFormRenderer from '@@ddf';
+import createSchema from './create-nsxt-cloud-network-form.schema';
+import { CloudNetworkApi } from '../../utils/cloud-network-api'
+import { ProvidersApi } from '../../utils/providers.api';
+import { handleApiError } from '../../utils/handle-api-error';
+
+class CreateNsxtCloudNetworkForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { loading: true };
+  }
+
+  componentDidMount() {
+    this.props.dispatch({
+      type: 'FormButtons.init',
+      payload: {
+        newRecord: true,
+        pristine: true
+      }
+    });
+    this.setInitialState();
+  }
+
+  setInitialState = async () => {
+    miqSparkleOn();
+    try {
+      const nsxt_provider = await ProvidersApi.find_nsxt_provider();
+      this.setState({ ems_id: nsxt_provider.id });
+    } catch (error) {
+      handleApiError(this, error);
+    }
+    this.setState({ loading: false });
+    miqSparkleOff();
+  }
+
+  submitValues = async (values) => {
+    miqSparkleOn();
+    try {
+      await CloudNetworkApi.create(values, this.state.ems_id);
+    } catch (error) {
```

Comment:
I'm not entirely sure if this error handling is necessary or should be done like this. If the API has an error, we have a default error handling around it that displays an error message anyway. So I'd just simplify this by displaying a higher-level error message in the [flash](https://github.com/ManageIQ/manageiq-providers-redfish/blob/master/app/javascript/components/redfish-server-firmware-update-dialog.jsx#L33-L44).